.gitignore

```diff
@@ -1,3 +1,4 @@
 input/
 output/
-.DS_Store
\ No newline at end of file
+.DS_Store
+.Rhistory
```

Comment:
So it's conventional to have a whiteline at the end of a file. You should be able to set your editor up to automatically add one when you save, e.g. `insert_final_newline = true` in an .editorconfig  file.

ChatGPT says:
The convention of having a white line at the end of code files is mainly related to the way certain programming languages handle the end of a file. In many programming languages, including C, C++, Java, and Python, the end of a file is indicated by an end-of-file (EOF) marker.

When a program reads a file, it typically reads until it reaches the EOF marker. However, if there is no EOF marker at the end of the file, the program may continue reading beyond the intended end of the file, which can lead to unexpected behavior or errors.

To prevent this from happening, it has become a convention to include a blank line at the end of code files. This ensures that there is an EOF marker at the end of the file, and that programs reading the file will stop at the intended end of the code.

While this convention is not strictly necessary in all programming languages or environments, it has become a common practice that can help prevent subtle bugs and ensure that code is read correctly by other programs.

Reply:
I don't think it is an important convention as no programming language really depends on it and doesn't change anything in code readability